RDKShell/RDKShell.cpp

```diff
@@ -271,10 +275,23 @@ static bool checkAssemblyFactoryMode_wrapper()
         bool ret = false;
         Device_Mode_Result_t result = Device_Mode_getFactoryMode(&factoryMode);
         if(result == DEVICE_MODE_RESULT_SUCCESS) {
+#ifdef RDKSHELL_DUAL_ODM_SUPPORT
```

Comment:
please help to move complete checks to a different function of getting factory app url.